### PR TITLE
regenesis: recompile handler

### PR DESCRIPTION
**Pull request by tynes**
**Description**
Adds the recompiler handler to the state surgery script. It includes a simple test but needs additional tests for contracts that have immutables and libraries.

### CODE REVIEW

File: packages/regenesis-surgery/scripts/utils.ts [hunk 1]
```diff
@@ -17,3 +21,57 @@ export const hexStringEqual = (a: string, b: string): boolean => {
...
+    sources: {
+      file: {
+        // TODO: does this need the brackets?
+        content: contract.sourceCode,
+      },
+    },
+    settings: {
+      outputSelection: {
+        '*': {
+          '*': ['*'],
```
Comment (elenadimitrova) [inline packages/regenesis-surgery/scripts/utils.ts:47]: Okay this includes `metadata` too I suppose. Now only thing is to preserve the `artifacts/build-info` for contracts to be uploaded to IPFS same as the Optimism contracts themselves see https://github.com/ethereum-optimism/optimism/pull/1534

**Comment by tynes** [comment]
Codechecks is now broken due to the diff it processes being too large

**Comment by tynes** [comment]
Will fix linting warnings

**Comment by tynes** [comment]
Codechecks is failing due to:

```
Executing 1 codechecks files
MaxBufferError: stdout maxBuffer exceeded
```

My guess is that its because the file of test vectors that I added is kind of large. Its contracts for mainnet, we could very easily add a test that iterates over each of them and calls the highest level function and just makes sure that it doesn't throw at all

**Pull request merged**